internal/openchoreo-api/services/component_service.go

```diff
@@ -1424,6 +1425,86 @@ func (s *ComponentService) GetComponent(ctx context.Context, orgName, projectNam
 	return s.toComponentResponse(component, typeSpecs, true), nil
 }
 
+// PatchComponent patches a Component with the provided updates
+func (s *ComponentService) PatchComponent(ctx context.Context, orgName, projectName, componentName string,
+	req *models.PatchComponentRequest) (*models.ComponentResponse, error) {
+	s.logger.Debug("Patching component", "org", orgName, "project", projectName, "component", componentName)
+
+	// Verify project exists
+	_, err := s.projectService.GetProject(ctx, orgName, projectName)
```

Comment:
we can remove this as this is done in https://github.com/openchoreo/openchoreo/pull/1040/files#diff-d81e6ca8ed7d2382b9f79c0da071993854e2d0f1b7e7dbe8b6b5173117465fa9R1457